addons/mail/models/mail_thread.py

```diff
@@ -3489,7 +3489,7 @@ def _notify_by_email_prepare_rendering_context(self, message, msg_vals=False,
         tracking = []
         if check_tracking:
             tracking_values = self.env['mail.tracking.value'].sudo().search(
-                [('mail_message_id', '=', message.id)]
+                [('mail_message_id', 'in', message.ids)]
```

Comment:
Having a distinction between `= .id` and `in .ids` for single record-set is very confusing.

Why can't newid be silently ignored instead of enforcing a strange syntax?

Reply:
You could probably update the condition on line 3490 to `if check_tracking and message.id:`, but in general, there are 
2 use cases. We do not want to handle an additional type of value since it's impossible to know what is the intention, we show a warning. In general, the fix is to use "in ids" (1st use-case).

1. `ids` returns the origin ids of NewId's. So it's not equivalent to ignoring the argument. You may want to search according to the original ids.
2. You may want to ignore the NewId altogether (and even avoid the search). You are changing the record, so you don't want to search on it.